### PR TITLE
:book: improve conversation errors and hook documentations

### PR DESCRIPTION
**Pull request by akirilyuk**

### CODE REVIEW

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 1]
```diff
@@ -136,9 +136,11 @@ Check for any interruption on Dialogflow side.
 #### Description
 This indicates an issue during Faas invocation.
 #### Reasons
-Mostly this happens if your quota limit is exceeded.
+This could have multiple reasons: function code non working, function invocation was not finished after 30s, Faas Quota limit was reached.
 #### Solutions
-Double-check you have enough api calls left for Faas.
+Double-check you have enough api calls left for Faas. Also please verify that your Faas function is working.
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:141]: Proposal of description of whole solution heading:

```
Please verify and ensure that you have enough API calls quota left for FaaS function and make sure that FaaS function is working and invokable. Refer to the [Faas Documentation](liveperson-functions-deployment.html#testing-your-function) to understand how to test your function code. 

Furthermore, please ensure your defined function follow the standard implementation as described in Third-Party bots documentation for [Custom Integration](third-party-bots-custom-integration.html). Failure to do so will result in the erroneous result.

```

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 2]
```diff
@@ -147,7 +149,7 @@ This happens if there is no response body provided.
 #### Reasons
 The Faas callback was invoked without a response body object.
 #### Solutions
-Provide a response body in the callback.
+Provide a response body in the callback, see [Custom Integration Documentation](liveperson-functions-deployment.html#testing-your-function)
```
Comment (fahimmahmoodmir) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:152]: `Please make sure to provide a response body in the callback, for further information see`

**Review by fahimmahmoodmir — CHANGES_REQUESTED** [review]
Did you already check with the examples for every vendor of Faas hooks? If not please verify them if they are working otherwise looks good

File: pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md [hunk 3]
```diff
@@ -186,7 +204,22 @@ Example Payloads:
...
 
-###### Lex
+###### Minimal working Faas example as code
+
+```js
+const { payload } = input;
+
+if (payload.message) {
+  input.payload.message += 'preHook';
+}
```
Comment (GFdiablo117) [inline pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md:214]: I know it's annoying, but could you change it from "input.payload.message" to "payload.message" on this example functions too?

File: pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md [hunk 4]
```diff
@@ -201,7 +234,20 @@ Example Payloads:
...
 
-###### Microsoft
+###### Minimal working Faas example as code
+
+```js
+const { payload } = input;
+if (payload.inputText) {
+  input.payload.inputText += 'preHook';
+}
+return callback(null, payload);
```
Comment (GFdiablo117) [inline pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md:244]: input.payload -> payload

File: pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md [hunk 5]
```diff
@@ -219,7 +265,19 @@ Example Payloads:
...
 
-###### WatsonV1
+###### Minimal working Faas example as code
+
+```js
+const { payload } = input;
+if (payload.text) {
+  input.payload.text += 'preHook';
+}
+return callback(null, payload);
```
Comment (GFdiablo117) [inline pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md:275]: input.payload -> payload

File: pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md [hunk 6]
```diff
@@ -244,13 +302,23 @@ Example Payloads:
...
 ```
 
+###### Minimal working Faas example as code
+
+```js
+const { payload } = input;
+if (payload.message) {
+  input.payload.message += 'preHook';
+}
+return callback(null, payload);
```
Comment (GFdiablo117) [inline pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md:312]: input.payload -> payload

File: pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md [hunk 7]
```diff
@@ -314,13 +382,43 @@ Example Responses:
...
+
+###### Minimal working Faas example as code  
+
+```js
+const { payload } = input;
+if (payload.messages && payload.messages[0]) {
+  input.payload.messages[0] += ' edited by postHook';
+}
+return callback(null, payload);
+```
```
Comment (GFdiablo117) [inline pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md:417]: input.payload -> payload

File: pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md [hunk 8]
```diff
@@ -335,7 +433,19 @@ Example Responses:
...
 
-###### Microsoft
+###### Minimal working Faas example as code 
+
+```js
+const { payload } = input;
+if (payload.message) {
+  input.payload.message = 'Changed text by postHook';
+}
+return callback(null, payload);
```
Comment (GFdiablo117) [inline pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md:443]: input.payload -> payload

File: pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md [hunk 9]
```diff
@@ -351,7 +461,22 @@ Example Responses:
...
+
+```js
+const { payload } = input;
+if (
+  payload &&
+  payload[0].text
+) {
+  input.payload[0].text += 'postHook changed this message';
+}
+return callback(null, payload);
```
Comment (GFdiablo117) [inline pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md:474]: input.payload -> payload

File: pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md [hunk 10]
```diff
@@ -390,4 +515,14 @@ Example Responses:
...
 }
+```
+
+###### Minimal working Faas example as code 
+
+```js
+const { payload } = input;
+if (payload.output && payload.output.generic[0].text === 'Hi there') {
+  input.payload.output.generic[0].text += 'postHook';
+}
```
Comment (GFdiablo117) [inline pages/Documents/ConversationalAI/Third-PartyBots/hook-configuration.md:526]: input.payload -> payload

File: pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md [hunk 11]
```diff
@@ -156,7 +158,8 @@ Indicates that the Faas hook could not be invoked.
 #### Reasons
 Implementation errors inside Faas hook.
 #### Solutions
-Double-check and test the Faas Hook code.
+Double-check and test the Faas Hook code. You can refer to the [Faas Documentation](liveperson-functions-deployment.html#testing-your-function) to find out how test your function code.
```
Comment (GFdiablo117) [inline pages/Documents/ConversationalAI/Third-PartyBots/bot-conversation-errors.md:161]: find out how test your function code -> find out how **to** test your function code

**Review by GFdiablo117 — CHANGES_REQUESTED** [review]

**Review by fahimmahmoodmir — APPROVED** [review]
approved from myside

**Pull request merged**